cpp/tests/strings/factories_test.cu

```diff
@@ -234,3 +236,200 @@ TEST_F(StringsFactoriesTest, StringPairWithNullsAndEmpty)
+  auto const stream           = cudf::get_default_stream();
+  auto const mr               = cudf::get_current_device_resource_ref();
+
+  std::vector<char const*> h_test_strings{"the quick brown fox jumps over the lazy dog",
+                                          "the fat cat lays next to the other accénted cat",
+                                          "a slow moving turtlé cannot catch the bird",
+                                          "which can be composéd together to form a more complete",
+                                          "thé result does not include the value in the sum in",
+                                          "",
+                                          nullptr,
+                                          "absent stop words"};
+  auto const num_test_strings = static_cast<int>(h_test_strings.size());
+
+  std::vector<std::size_t> h_offsets(num_test_strings + 1, 0);
+  for (int i = 0; i < num_test_strings; ++i) {
+    h_offsets[i + 1] = h_offsets[i] + (h_test_strings[i] ? strlen(h_test_strings[i]) : 0);
+  }
+
+  std::vector<char> h_chars(h_offsets.back());
+  std::vector<int> is_null(num_test_strings, 0);
+  for (int i = 0; i < num_test_strings; ++i) {
+    if (h_test_strings[i]) {
+      memcpy(h_chars.data() + h_offsets[i], h_test_strings[i], strlen(h_test_strings[i]));
+    } else {
+      is_null[i] = 1;
+    }
+  }
+
+  auto const d_offsets = cudf::detail::make_device_uvector_async(h_offsets, stream, mr);
+  auto const d_chars   = cudf::detail::make_device_uvector_async(h_chars, stream, mr);
+  auto const d_is_null = cudf::detail::make_device_uvector_async(is_null, stream, mr);
+
+  std::vector<rmm::device_uvector<string_pair>> d_input;
+  std::vector<cudf::device_span<string_pair const>> input;
+  d_input.reserve(num_columns);
+  input.reserve(num_columns);
+
+  for (int col_idx = 0; col_idx < num_columns; ++col_idx) {
+    // Columns have sizes increase from `max_num_rows / num_columns` to `max_num_rows`.
+    auto const num_rows =
+      static_cast<int>(static_cast<double>(col_idx + 1) / num_columns * max_num_rows);
+
+    auto string_pairs = rmm::device_uvector<string_pair>(num_rows, stream);
+    thrust::tabulate(
+      rmm::exec_policy_nosync(stream),
+      string_pairs.begin(),
+      string_pairs.end(),
+      index_to_pair{num_test_strings, d_chars.begin(), d_offsets.begin(), d_is_null.begin()});
+
+    d_input.emplace_back(std::move(string_pairs));
+    input.emplace_back(d_input.back());
+  }
+
+  auto const output = cudf::make_strings_column_batch(input, stream, mr);
+
+  for (std::size_t i = 0; i < num_columns; ++i) {
+    auto const string_pairs = input[i];
+    auto const expected     = cudf::make_strings_column(string_pairs, stream, mr);
+    CUDF_TEST_EXPECT_COLUMNS_EQUAL(expected->view(), output[i]->view());
+  }
+}
+
+// The test below requires a huge amount of memory, thus it is disabled by default.
+#ifdef LOCAL_TEST
+TEST_F(StringsBatchConstructionTest, CreateLongStringsColumns)
```

Comment:
```suggestion
TEST_F(StringsBatchConstructionTest, DISABLED_CreateLongStringsColumns)
```
You can add `DISABLED_` prefix to any test to disable it. This is a little bit cleaner or more consistent than adding a new `#define`.